force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -122,20 +134,18 @@ public with sharing class ClassesObjectsMethods {
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        String resultString = '';
+
+        for(String s: scientistNames) {
+            String str = s.toLowerCase();
+            if(str.containsNone('g')) {
+                continue;
+            }
+            resultString += s + ',';    
```

Comment:
How could we make this code more efficient? Do we need to check for none and then continue? Or could we just cut to the chase and add the name whenever it contains the letter 'g'? Consider using the `contains()` or `containsIgnoreCase()` method here.